jobspec/parse_service.go

```diff
@@ -228,6 +228,54 @@ func parseSidecarService(o *ast.ObjectItem) (*api.ConsulSidecarService, error) {
 	return &sidecar, nil
 }
 
+func parseSidecarTask(item *ast.ObjectItem) (*api.SidecarTask, error) {
+	// We need this later
+	var listVal *ast.ObjectList
+	if ot, ok := item.Val.(*ast.ObjectType); ok {
+		listVal = ot.List
+	} else {
+		return nil, fmt.Errorf("should be an object")
+	}
+
+	// Check for invalid keys
+	valid := []string{
+		"config",
+		"driver",
+		"env",
+		"kill_timeout",
+		"logs",
+		"meta",
+		"resources",
+		"shutdown_delay",
+		"user",
+		"kill_signal",
+	}
+	if err := helper.CheckHCLKeys(listVal, valid); err != nil {
+		return nil, err
+	}
+
+	task, err := parseTask(item)
```

Comment:
Ha, I never thought about it but with `CheckHCLKeys + parseSuperset` - adding one off structs that are a subset of another struct is super easy. You got lucky. :sweat_smile:

Reply:
I tried another trick to keep from having to copy all of parseTask for handling ShutdownDelay.